components/dialog/dialog-mixin.js

```diff
@@ -0,0 +1,286 @@
+import '../backdrop/backdrop.js';
+import { findComposedAncestor, isComposedAncestor } from '../../helpers/dom.js';
+import { getComposedActiveElement, getNextFocusable, getPreviousFocusable } from '../../helpers/focus.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { html } from 'lit-element/lit-element.js';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+import { styleMap } from 'lit-html/directives/style-map.js';
+
+export const DialogMixin = superclass => class extends RtlMixin(superclass) {
+
+	static get properties() {
+		return {
+			opened: { type: Boolean, reflect: true },
+			titleText: { type: String, attribute: 'title-text' },
+			_height: { type: Number },
+			_left: { type: Number },
+			_margin: { type: Object },
+			_nestedShowing: { type: Boolean },
+			_overflowBottom: { type: Boolean },
+			_overflowTop: { type: Boolean },
+			_parentDialog: { type: Object },
+			_state: { type: String, reflect: true },
+			_width: { type: Number }
+		};
+	}
+
+	constructor() {
+		super();
+		this.opened = false;
+		this._dialogId = getUniqueId();
+		this._height = 0;
+		this._margin = { top: 100, right: 30, bottom: 30, left: 30 };
+		this._parentDialog = null;
+		this._nestedShowing = false;
+		this._state = null;
+		this._left = 0;
+		this._width = 0;
+		//this._hasNativeDialog = (window.HTMLDialogElement !== undefined);
```

Comment:
We'd discussed exposing this as a hidden property that you could just set to force native dialogs to use the polyfill. That way your visual diffs running in Chrome could test both types.